pkg/addons/vpc_controller.go

```diff
@@ -134,15 +136,53 @@ func (v *VPCController) generateCert() error {
 		return errors.Wrap(err, "updating approval")
 	}
 
-	approvedCSR, err := csrClientSet.Get(certificateSigningRequest.Name, metav1.GetOptions{})
+	logger.Info("waiting for certificate to be available")
+
+	watchTimeout := 45 * time.Second
+	cert, err := watchCSRApproval(csrClientSet, csrName, watchTimeout)
 	if err != nil {
 		return err
 	}
 
-	if approvedCSR.Status.Certificate == nil {
-		return errors.New("failed to find certificate after approval")
+	return v.createCertSecrets(privateKey, cert)
+}
+
+func watchCSRApproval(csrClientSet v1beta1.CertificateSigningRequestInterface, csrName string, timeout time.Duration) ([]byte, error) {
+	watcher, err := csrClientSet.Watch(metav1.ListOptions{
+		FieldSelector: fmt.Sprintf("metadata.name=%s", csrName),
+	})
+
+	if err != nil {
+		return nil, err
 	}
-	return v.createCertSecrets(privateKey, approvedCSR.Status.Certificate)
+
+	defer watcher.Stop()
+
+	timer := time.NewTimer(timeout)
+	defer timer.Stop()
+
+	for {
+		select {
+		case event, ok := <-watcher.ResultChan():
+			if !ok {
+				return nil, errors.New("unexpected close of ResultChan")
```

Comment:
Might not be super useful to the user. Perhaps "failed waiting for certificate. unexpected close of channel"?